### PR TITLE
Add register API endpoint

### LINKED ISSUE #62: Add /register API endpoint
Needs POST request.

### PR DESCRIPTION
**Pull request by Herover**
Closes #62

### CODE REVIEW

File: minitwit/src/main/java/WebApplication.java [hunk 1]
```diff
@@ -556,4 +570,22 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
...
+        var error = register(username, email, password, password);
+        if (error != null) {
+            response.status(400);
+            return Map.ofEntries(
+                    Map.entry("status", 400),
+                    Map.entry("error_msg", error)
+            );
+        }
+
+        return Map.ofEntries();
```
Comment (Herover) [inline minitwit/src/main/java/WebApplication.java:589]: The spec just returns a empty body but I couldn't figure out a neat way to do that with out gson setup, so now we return a empty object as body...

File: minitwit/src/main/java/WebApplication.java [hunk 2]
```diff
@@ -556,4 +570,22 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
...
         return filteredMessages.toList();
     };
+
+    public static Route serveSimRegister = (Request request, Response response) -> {
+        var json = gson.fromJson(request.body(), HashMap.class);
+        var username = String.valueOf(json.get("username"));
+        var email = String.valueOf(json.get("email"));
+        var password = String.valueOf(json.get("pwd"));
+
+        var error = register(username, email, password, password);
```
Comment (smilladion) [inline minitwit/src/main/java/WebApplication.java:580]: Does the simulator only provide one password, as opposed to the UI where you need to repeat your password, thus providing two? Bit strange, but if they made it like this, then I suppose what you did here is correct. 😛

File: minitwit/src/main/java/WebApplication.java (hunk 1 above)
Comment (smilladion) [reply minitwit/src/main/java/WebApplication.java:589]: I think that's fine, but looks like the Python method returns 204 as a status code as well as the empty body. Maybe we need that here too? Not sure where we specify status codes in our program.

**Comment by smilladion** [comment]
You might need to merge this with `latest-endpoint` as every API route needs to call `updateLatest(request)`!

File: minitwit/src/main/java/WebApplication.java (hunk 1 above)
Comment (Herover) [reply minitwit/src/main/java/WebApplication.java:589]: Reading the simulators code it looks like the exact return codes are important so I'll try figure something out.

File: minitwit/src/main/java/WebApplication.java (hunk 2 above)
Comment (Herover) [reply minitwit/src/main/java/WebApplication.java:580]: Yesp!

**Review by smilladion — APPROVED** [review]
Gives me the correct status code now, and no errors when using `curl`. So I believe it works!

**Pull request merged**